python_modules/dagstermill/dagstermill/__init__.py

```diff
@@ -173,65 +161,25 @@ def serialize_dm_context(transform_execution_info, inputs):
     if not os.path.exists(marshal_dir):
         os.makedirs(marshal_dir)
 
-    new_inputs_structure = {
+    dm_context_dict = {
         'pipeline_name': transform_execution_info.pipeline_def.name,
         'solid_def_name': transform_execution_info.solid.definition.name,
-        'solid_name': transform_execution_info.solid.name,
-        'inputs': {},
         'marshal_dir': marshal_dir,
     }
 
+    parameters = dict(dm_context=json.dumps(dm_context_dict))
+
     input_defs = transform_execution_info.solid_def.input_defs
     input_def_dict = {inp.name: inp for inp in input_defs}
     for input_name, input_value in inputs.items():
+        assert input_name != "dm_context", "Protected input name!"
```

Comment:
"... Dagstermill solids may not be defined with the input 'dm_context'" (tell the user what to change)